src/schema/objects/formats.yaml

```diff
@@ -47,9 +47,15 @@ hed_version:
   display_name: HED Version
   description: |
     The version string of the used HED schema.
-  pattern: '^(?:[a-zA-Z]+:)?(?:[a-zA-Z]+_)?(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\.(?:0|[1-9]\d*)\
-    (?:-(?:(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?\
-    (?:\+(?:[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$'
+  pattern: "\
+    (?:[a-zA-Z]+:)?\
+    (?:[a-zA-Z]+_)?\
+    (?:0|[1-9][0-9]*)\\.(?:0|[1-9][0-9]*)\\.(?:0|[1-9][0-9]*)\
+    (?:-(?:\
+      (?:0|[1-9][0-9]*|[0-9]*[a-zA-Z-][0-9a-zA-Z\\-]*)\
+      (?:\\.(?:0|[1-9][0-9]*|[0-9]*[a-zA-Z\\-][0-9a-zA-Z\\-]*))*\
+    ))?\
+    (?:\\+(?:[0-9a-zA-Z\\-]+(?:\\.[0-9a-zA-Z\\-]+)*))?"
```

Comment:
@happy5214 Can you verify that this pattern is correct?

Changes:

1) `\d` -> `[0-9]`
2) Lone `-` inside `[]` becomes `\\-`
3) Extra newlines to help break up logical chunks.

[Renders](https://bids-specification--2077.org.readthedocs.build/en/2077/schema.json) as:

```json
"(?:[a-zA-Z]+:)?(?:[a-zA-Z]+_)?(?:0|[1-9][0-9]*)\\.(?:0|[1-9][0-9]*)\\.(?:0|[1-9][0-9]*)(?:-(?:(?:0|[1-9][0-9]*|[0-9]*[a-zA-Z-][0-9a-zA-Z\\-]*)(?:\\.(?:0|[1-9][0-9]*|[0-9]*[a-zA-Z\\-][0-9a-zA-Z\\-]*))*))?(?:\\+(?:[0-9a-zA-Z\\-]+(?:\\.[0-9a-zA-Z\\-]+)*))?"
```

Reply:
I couldn't even verify the *original* was correct. I only contributed the first two groups (the ones ending with the colon and underscore) and copied the rest from https://semver.org/. @VisLab Can a HED schema have any valid semantic version, or are there limits within that?

Reply:
We should only use plain x.y.z.   I don't think the specification says anything to the contrary. There are a few odd things in the deprecated folder like HED8.0.0-alpha.1.xml.  These have never been released and no known datasets use them.  (Validators don't support these preliminary formats.) I think we should remove them. 

Schemas in the prerelease folder also conform to x.y.z versioning.

Reply:
To clarify, the first two "lines" (in the updated version) are the prefix and library name, and those should be kept even if the version number is simplified.

Reply:
Just reading this, the following is permissible:

```
xYz:HjK_10.0.3-7a4Q-0-0-0-.8b5R-1-1-1-+asduijkqr98926ah1.asduijkqr98926ah1.asduijkqr98926ah1
|--||--||----|||---------||----------||----------------------------------------------------|  
 L1  L2   L3  4    L5          L6                            L7
```

But I agree that the semver regex appears to allow this, for everything after `xYz:HjK_`, so if you're happy with this, we can call it good.

Reply:
Our rule is that "official" schema releases use only `x.y.z` version numbering (pre-releases, which pre-8.0.0 followed the `-alpha` tagging scheme mentioned by @VisLab, now share the same version numbering as official releases; either way, they should not be used). Therefore, anything after the `<prefix>:<library>_x.y.z` should be removed entirely.

Reply:
```suggestion
  pattern: "(?:[a-zA-Z]+:)?(?:[a-zA-Z]+_)?(?:0|[1-9][0-9]*)\\.(?:0|[1-9][0-9]*)\\.(?:0|[1-9][0-9]*)"
```